python/ray/cloudpickle/cloudpickle_fast.py

```diff
@@ -508,6 +552,13 @@ class CloudPickler(Pickler):
 
 
     dispatch_table = ChainMap(_dispatch_table, copyreg.dispatch_table)
+
+    if importlib.util.find_spec("pydantic") is not None:
+        dispatch_table[ModelField] = _pydantic_model_field_reduce
```

Comment:
should this be `pydantic.fields.ModelField`? How can we ensure `pydantic` is in scope here as well?

Reply:
You have to import pydantic here. The best practice should be something like in https://github.com/ray-project/ray/pull/12344 as we discussed earlier

Reply:
Agreed, the issue with the add-ons right now is that the ray client isn't using the same serialization codepath so it won't work for ray client (cc @AmeerHajAli)